commons/src/main/java/nl/nn/adapterframework/util/DateFormatUtils.java

```diff
@@ -44,28 +45,40 @@ public class DateFormatUtils {
 
 
 	public static String now() {
-		return format(new Date());
+		return format(Instant.now());
 	}
 
 	@Deprecated
 	public static String now(String format) {
-		return format(new Date(), new SimpleDateFormat(format));
+		return format(Instant.now(), new SimpleDateFormat(format));
 	}
 
-	public static String format(long date) {
-		return format(new Date(date));
+	public static String format(Date date) {
+		return format(date.getTime());
 	}
 
-	public static String format(Date date) {
+	public static String format(long date) {
 		return format(date, FULL_GENERIC_FORMATTER);
 	}
 
-	public static String format(Date date, DateFormat formatter) {
+	public static String format(Instant instant) {
+		return format(instant.toEpochMilli(), FULL_GENERIC_FORMATTER);
+	}
+
+	public static String format(long date, DateFormat formatter) {
 		return formatter.format(date);
 	}
 
+	public static String format(long date, String format) {
+		return format(date, new SimpleDateFormat(format));
+	}
+
 	public static String format(Date date, String dateFormat) {
-		return format(date, new SimpleDateFormat(dateFormat));
+		return format(date.getTime(), new SimpleDateFormat(dateFormat));
+	}
+
+	public static String format(Instant instant, DateFormat formatter) {
+		return format(instant.toEpochMilli(), formatter);
```

Comment:
Is it possible to use `java.time.format.DateTimeFormatter` instead of the old-fashioned and non-thread-safe `java.text.DateFormat` classes?
At least have an option to use those for all calls that pass the format as a String, and perhaps maintain the old formatter for legacy code that really can't be changed now to use a "modern" API?

The modern API formatters can also work directly with an `Instant` instead of having to get the millis out first.